tests/kubernetes/minikube.py

```diff
@@ -0,0 +1,204 @@
+            else:
+                yamls = sorted([os.path.join(services_dir, y) for y in yamls if y.endswith(".yaml")])
+        print()
+        for y in yamls:
+            body = yaml.load(open(y))
+            kind = body["kind"]
+            name = body['metadata']['name']
+            namespace = body['metadata']['namespace']
+            if "configmap" in y:
+                if has_configmap(name, namespace=namespace):
+                    print("Deleting configmap \"%s\" ..." % name)
+                    delete_configmap(name, namespace=namespace)
+                print("Creating configmap from %s ..." % y)
+                create_configmap(body=yaml.load(open(y)))
+                self.yamls.append(body)
+        for y in yamls:
+            body = yaml.load(open(y))
+            kind = body["kind"]
+            name = body['metadata']['name']
+            namespace = body['metadata']['namespace']
+            body = yaml.load(open(y))
+            if kind == "ConfigMap":
+                continue
+            assert kind == "Deployment", "kind \"%s\" in %s not yet supported!" % (kind, y)
+            if has_deployment(name, namespace=namespace):
+                print("Deleting deployment \"%s\" ..." % name)
+                delete_deployment(name, namespace=namespace)
+            print("Creating deployment from %s ..." % y)
+            create_deployment(body=body)
+            self.yamls.append(body)
+        if len(self.yamls) > 0:
+            assert wait_for(all_pods_have_ips, timeout_seconds=300), "timed out waiting for pod IPs!"
+        try:
+            yield
+        finally:
+            for y in self.yamls:
+                kind = y["kind"]
+                name = y['metadata']['name']
+                namespace = y['metadata']['namespace']
+                if kind == "ConfigMap":
+                    print("Deleting configmap \"%s\" ..." % name)
+                    delete_configmap(name, namespace=namespace)
+                elif kind == "Deployment":
+                    print("Deleting deployment \"%s\" ..." % name)
+                    delete_deployment(name, namespace=namespace)
+            self.yamls = []
+
+
+    def pull_agent_image(self, name, tag="latest"):
+        host_client = docker.from_env(version="auto")
+        try:
+            image_id = host_client.images.get("%s:%s" % (name, tag)).id
+        except:
+            image_id = None
+        if image_id:
+            try:
+                self.client.images.get(image_id)
+                return
+            except:
+                pass
+        print("\nPulling %s:%s to the minikube container ..." % (name, tag))
+        self.container.exec_run("cp -f /etc/hosts /etc/hosts.orig")
+        self.container.exec_run("cp -f /etc/hosts /etc/hosts.new")
+        self.container.exec_run("sed -i 's|127.0.0.1|%s|' /etc/hosts.new" % get_host_ip())
```

Comment:
This is pretty hacky, what is this trying to do?  We can probably figure out a better way to do this.

Reply:
The registry container used to get the agent image to the minikube container will only listen on localhost (unless we use TLS). So this just temporarily routes localhost to the host IP that the registry container is running on so that we can pull agent image.